javaagent-extension-api/src/main/java/io/opentelemetry/javaagent/bootstrap/internal/InstrumentationConfig.java

```diff
@@ -5,37 +5,24 @@
 
 package io.opentelemetry.javaagent.bootstrap.internal;
 
-import static java.util.Collections.emptyList;
 import static java.util.Objects.requireNonNull;
 
-import java.time.Duration;
-import java.util.List;
-import java.util.Map;
+import io.opentelemetry.instrumentation.api.incubator.config.internal.CoreInstrumentationConfig;
 import java.util.logging.Logger;
-import javax.annotation.Nullable;
 
 /**
- * Represents the global instrumentation configuration consisting of system properties, environment
- * variables, contents of the agent configuration file and properties defined by the {@code
- * ConfigPropertySource} SPI implementations.
- *
- * <p>In case any {@code get*()} method variant gets called for the same property more than once
- * (e.g. each time an advice class executes) it is suggested to cache the result instead of
- * repeatedly calling {@link InstrumentationConfig}. Instrumentation configuration does not change
- * during the runtime so retrieving the property once and storing its result in a static final field
- * allows JIT to do its magic and remove some code branches.
- *
- * <p>This class is internal and is hence not for public use. Its APIs are unstable and can change
- * at any time.
+ * This class is internal and is hence not for public use. Its APIs are unstable and can change at
+ * any time.
  */
-public abstract class InstrumentationConfig {
+public class InstrumentationConfig {
```

Comment:
wdyt if we renamed these to `AgentInstrumentationConfig` and `AgentCommonConfig` then instead of `CoreInstrumentationConfig` we could use `InstrumentationConfig`.

Reply:
great idea - I'll do the same for CommonConfig